src/olympia/templates/copyright.html

```diff
@@ -22,9 +22,9 @@
   <p>
     {% trans %}
       Except where otherwise
-      <a href="http://www.mozilla.org/about/legal.html#site">noted</a>,
+      <a href="https://www.mozilla.org/about/legal/#site">noted</a>,
```

Comment:
I'm guessing it's been moved to https://www.mozilla.org/en-US/about/legal/terms/mozilla/, which seems specifically about websites.